lib/rex/post/meterpreter/extension_mapper.rb

```diff
@@ -16,18 +16,24 @@ def self.get_extension_names
   end
 
   def self.get_extension_id(name)
-    k = self.get_extension_klass(name)
+    begin
+      k = self.get_extension_klass(name)
+    rescue RuntimeError
+      return nil
+    end
+
     k.extension_id
   end
 
   def self.get_extension_name(id)
-    self.get_extension_names.each do |name|
+    self.get_extension_names.find do |name|
       begin
         klass = self.get_extension_klass(name)
       rescue RuntimeError
         next
       end
-      return name if klass.extension_id == id
+
+      klass.extension_id == id
```

Comment:
These changes here update these two methods to both return `nil` when the ID or name is unknown. Previously id would raise a RuntimeError and name would return a list. This allows them to be used to easily query whether or not the ID and name are valid which I was doing while experimenting with some ways to suggest extensions. The `get_extension_module` and `get_extension_klass` method both still raise `RuntimeError` when the name is invalid.